Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -153,3 +160,222 @@ theorem LSeries_eq_mul_integral_of_nonneg (f : ℕ → ℝ) {r : ℝ} (hr : 0 
+    exact mul_le_mul_of_nonneg_right (hS₂ t ht) (norm_nonneg _)
+  · exact integrableAtFilter_rpow_atTop_iff.mpr <| neg_lt_neg_iff.mpr hs
+
+end lemmas
+
+section proof
+
+include hlim in
+private theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₁ {ε : ℝ} (hε : ε > 0) :
+    ∀ᶠ t : ℝ in atTop, ‖(∑ k ∈ Icc 1 ⌊t⌋₊, f k) - l * t‖ < ε * t := by
+  have h_lim' : Tendsto (fun t : ℝ ↦ (∑ k ∈ Icc 1 ⌊t⌋₊, f k : ℂ) / t) atTop (𝓝 l) := by
+    refine (mul_one l ▸ ofReal_one ▸ ((hlim.comp tendsto_nat_floor_atTop).mul <|
+      tendsto_ofReal_iff.mpr <| tendsto_nat_floor_div_atTop)).congr' ?_
+    filter_upwards [eventually_ge_atTop 1] with t ht
+    rw [Function.comp_def, ofReal_div, ofReal_natCast, div_mul_div_cancel₀ (by
+      rwa [Nat.cast_ne_zero, ne_eq, Nat.floor_eq_zero, not_lt])]
+  rw [Metric.tendsto_nhds] at h_lim'
+  filter_upwards [eventually_gt_atTop 0, h_lim' ε hε] with t ht₁ ht₂
+  rwa [dist_eq_norm, div_sub' _ _ _ (ne_zero_of_re_pos ht₁), norm_div, norm_real,
+    Real.norm_of_nonneg ht₁.le, mul_comm, div_lt_iff₀ ht₁] at ht₂
+
+private theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₂ {s T ε : ℝ} {S : ℝ → ℂ}
+    (hS : LocallyIntegrableOn (fun t ↦ S t - l * t) (Set.Ici 1)) (hε : 0 < ε)
+    (hs : 1 < s) (hT₁ : 1 ≤ T) (hT : ∀ t ≥ T, ‖S t - l * t‖ ≤ ε * t) :
+    (s - 1) * ∫ (t : ℝ) in Set.Ioi T, ‖S t - l * t‖ * t ^ (-s - 1) ≤ ε := by
+  have hT₀ : 0 < T := zero_lt_one.trans_le hT₁
+  have h {t : ℝ} (ht : 0 < t) : t ^ (-s) = t * t ^ (-s - 1) := by
+    rw [Real.rpow_sub ht, Real.rpow_one, mul_div_cancel₀ _ ht.ne']
+  calc
+    _ ≤ (s - 1) * ∫ (t : ℝ) in Set.Ioi T, ε * t ^ (-s) := ?_
+    _ ≤ ε * ((s - 1) * ∫ (t : ℝ) in Set.Ioi 1, t ^ (-s)) := ?_
+    _ = ε := ?_
+  · refine mul_le_mul_of_nonneg_left (setIntegral_mono_on ?_ ?_ measurableSet_Ioi fun t ht ↦ ?_) ?_
+    · exact (lemma₂ hs hT₀ hS hT).mono_set <| Set.Ioi_subset_Ici_iff.mpr hT₁
+    · exact (integrableOn_Ioi_rpow_of_lt (neg_lt_neg_iff.mpr hs) hT₀).const_mul  _
+    · have ht' : 0 < t := hT₀.trans ht
+      rw [h ht', ← mul_assoc]
+      exact mul_le_mul_of_nonneg_right (hT t ht.le) (Real.rpow_nonneg ht'.le _)
+    · exact (sub_pos_of_lt hs).le
+  · rw [integral_mul_left, ← mul_assoc, ← mul_assoc, mul_comm ε]
+    refine mul_le_mul_of_nonneg_left (setIntegral_mono_set ?_ ?_
+      (Set.Ioi_subset_Ioi hT₁).eventuallyLE) (mul_nonneg (sub_pos_of_lt hs).le hε.le)
+    · exact integrableOn_Ioi_rpow_of_lt (neg_lt_neg_iff.mpr hs) zero_lt_one
+    · exact (ae_restrict_iff' measurableSet_Ioi).mpr <| univ_mem' fun t ht ↦
+        Real.rpow_nonneg (zero_le_one.trans ht.le) _
+  · rw [integral_Ioi_rpow_of_lt (by rwa [neg_lt_neg_iff]) zero_lt_one, Real.one_rpow, neg_div,
+      ← one_div_neg_eq_neg_one_div, neg_add', neg_neg, mul_one_div,
+      div_self (sub_ne_zero.mpr hs.ne'), mul_one]
+
+variable (hfS : ∀ s : ℝ, 1 < s → LSeriesSummable f s)
+
+include hlim hfS in
+private theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₃ {ε : ℝ} (hε : ε > 0) :
```

Comment:
This proof is very long and needs some comments to the strategy a bit clearer. (I've heard it suggested that you should have at least one comment every 10 lines.)

Reply:
David, I added some comments to explain the proof. I also think I addressed all your suggestions. Thanks again for the review!